pkg/reconciler/route/route.go

```diff
@@ -483,3 +489,19 @@ func setTargetsScheme(rs *v1alpha1.RouteStatus, dnsNames []string, scheme string
 		}
 	}
 }
+
+func certMatches(domains []string, cert *netv1alpha1.Certificate) bool {
+	dnsNames := sets.NewString()
+
+	for _, dns := range cert.Spec.DNSNames {
```

Comment:
can we be sure this won't panic?

Reply:
I think so. `cert` can't be nil because of the check on line 317. `Spec` isn't a pointer so it also won't be nil and we can rely on the zero values of its fields here, in this case `DNSNames` will just be an empty slice.